pkg/controller/v1beta1/trainedmodel/shard/shard_manager.go

```diff
@@ -0,0 +1,21 @@
+package shard
+
+import (
+	v1beta1api "github.com/kubeflow/kfserving/pkg/apis/serving/v1beta1"
+)
+
+type ShardingStrategy string
+
+const (
+	Memory ShardingStrategy = "Memory"
+)
+
+type ShardManager struct {
+	Strategy ShardingStrategy
+}
+
+// Return a TrainedModel's shardId
+func (v *ShardManager) GetShardIdForTrainedModel(trainedModel *v1beta1api.TrainedModel) int {
```

Comment:
I'd expect this interface to look something like
```
strategy.GetOrAssignShard(trainedModel, inferenceService)
```

Reply:
Sounds good, I only passed in trainedModel for now since this we don't have v1beta1.InferenceService api in this pr